misk/src/main/kotlin/misk/digester/MergingDigest.kt

```diff
@@ -0,0 +1,323 @@
+package misk.digester
+
+import tdigest.Centroid
+import tdigest.MergingDigestData
+import kotlin.math.PI
+import kotlin.math.asin
+
+/**
+ * Provides an implementation of t-digest to be used to send histogram data
+ *
+ * Direct port of Veneur digest created by Stripe
+ * https://github.com/stripe/veneur/blob/master/tdigest/merging_digest.go
+ */
+class MergingDigest {
+
+  // amount of compression to data
+  private val compression: Double
+
+  // main list of centroids
+  private var mainCentroids: MutableList<Centroid> = mutableListOf()
```

Comment:
Why is this both a `var` and `mutable`? Should likely either be a mutable `val` (if you are changing the contents of the list in place) or a `var) (if you are changing the reference to the list but not updating the contents in place)

Reply:
Both properties are used:
`mainCentroids = mergingDigestData.main_centroids.toMutableList()`
`mainCentroids.add(next)`